sklearn/metrics/_classification.py

```diff
@@ -48,11 +48,15 @@
 
 def _check_zero_division(zero_division):
     if isinstance(zero_division, str) and zero_division == "warn":
-        return
+        return np.float64(0.0)
     elif isinstance(zero_division, (int, float)) and zero_division in [0, 1]:
-        return
+        return np.float64(zero_division)
+    elif np.isnan(zero_division):
+        return np.float64(np.nan)
```

Comment:
I think `np.nan` is already a `np.float64`. I don't think that we need to pass it in the constructor.